src/backend/distributed/planner/shard_pruning.c

```diff
@@ -803,6 +818,19 @@ AddPartitionKeyRestrictionToInstance(ClauseWalkerContext *context, OpExpr *opCla
 
 			case ROWCOMPARE_NE:
 			{
+				/*
```

Comment:
I couldn't follow this change. Is this related to the fix?

Reply:
No, not directly; just while reading the logic I saw that `ROWCOMPARE_NE` was some sort of magic value returned because PostgreSQL _consider[s] a `<>` operator to be a "member" of an opfamily if its negator is an equality operator of the opfamily_ and returns _ROWCOMPARE_NE […] for this case_.

Since we're iterating over a list of these things, either (a) the list should contain no `ROWCOMPARE_NE` elements, or (b) it should contain _only_ `ROWCOMPARE_NE` elements. But _nothing_ in our code actually reflected this behavior of the underlying code, and I was worried enough about further correctness issues in this pruning stuff that I threw in some logic to `Assert` or short-circuit in those cases (we don't understand `ROWCOMPARE_NE` at anyhow).